uvicorn/server.py

```diff
@@ -240,12 +239,7 @@ async def shutdown(self, sockets=None):
             await self.lifespan.shutdown()
 
     def install_signal_handlers(self) -> None:
-        if threading.current_thread() is not threading.main_thread():
-            # Signals can only be listened to from the main thread.
-            return
```

Comment:
We added this via #871, and I don't think we should be dropping it. Sure, maybe we don't really care about it in our own test suite anymore, but people who are running Uvicorn within threads certainly still do. Right?

Reply:
Oo, I never think about that, I reverted